File: prophecies/core/models/task_checker.py
from django.contrib.auth.models import User
from django.db import models

from prophecies.core.models.task import Task


class TaskChecker(models.Model):
    checker = models.ForeignKey(User, on_delete=models.CASCADE)
    task = models.ForeignKey(Task, on_delete=models.CASCADE)

    def __str__(self):
        return f'{self.checker} is checker on task #{self.task}'

    class Meta:
        unique_together = ('checker', 'task')

File: prophecies/apps/frontend/src/models/TaskChecker.js
import { Model } from '@vuex-orm/core'
import Task from '@/models/Task'
import User from '@/models/User'

export default class TaskChecker extends Model {
  static entity = 'TaskChecker'
  static primaryKey = ['taskId', 'checkerId']

  static fields () {
    return {
      id: this.attr(null),
      checkerId: this.attr(null),
      checker: this.belongsTo(User),
      taskId: this.attr(null),
      task: this.belongsTo(Task)
    }
  }
}
